### DISCUSSION TITLE
inputNumber comma is not working in numeric keypad

### DISCUSSION THREAD

**Discussion by itnoldev** [primefaces/community#393] [PrimeFaces] [answered]
I'm working with the Primefaces version 12.0.4. 
Comma is not working when I input numbers with numeric keypad.

This is my faces-config file: 

```
<application>
    <locale-config>
        <default-locale>it_IT</default-locale>
        <supported-locale>en_US</supported-locale>
        <supported-locale>it_IT</supported-locale>
    </locale-config>
    <currency-code>EUR</currency-code>
    <number-grouping-separator>.</number-grouping-separator>
    <decimal-separator>,</decimal-separator>
</application>
```

I had no problem with older version of PF. There is a way to make it works?

**Comment by melloware** [comment]
Are you using InputNumber?  Where are you saying its "not working" ?

**Reply by itnoldev** [reply]
I'm using p:inputNumber.
The comma on the numpad of the keyboard is not working.
As you can see from the faces-config file, I'm using ',' as the decimal separator. When I input numbers with the numpad and press the dot (.), nothing happens.

**Reply by melloware** [reply]
@itnoldev maybe I am confused but these two settings...

```xml
<number-grouping-separator>.</number-grouping-separator>
<decimal-separator>,</decimal-separator>
```

Are not valid in faces-config.xml do you have docs that say otherwise? 

https://javaee.github.io/javaserverfaces/docs/2.3/javadocs/web-facesconfig.html#type_faces-config-applicationType

I have no idea how this worked in older PrimeFaces but it should NOT have.  InputNumber reads those values from your `locale` see docs https://primefaces.github.io/primefaces/13_0_0/#/components/inputnumber?id=inputnumber  so it should be using your `it_IT`locale.

can you create a PrimeFaces Test reproducer showing the issue?

**Reply by itnoldev** [reply]
Ok. I deleted the separators tags, but my faces-config file have locale setting:

```
<application>
    <locale-config>
        <default-locale>it_IT</default-locale>
        <supported-locale>en_US</supported-locale>
        <supported-locale>it_IT</supported-locale>
    </locale-config>
</application>
```
I'm using jakarta faces 4.0.1

The inputNumber documentation say that decimalSeparator and thousandSeparator "Depends on locale defined via faces-config.xml".
So I think my settings are good.

**Reply by melloware** [reply]
ok let me try.

**Reply by melloware** [reply]
@itnoldev did you upgrade your JDK too?  Because here is a reproducer: 
[pf-inputnumber-it.zip](https://github.com/primefaces/community/files/13269846/pf-inputnumber-it.zip)

Run `mvn clean jetty:run` and navigate to http://localhost:8080/primefaces-test/ you will see with locale `de_DE` it properly uses the local separators.  With `it_IT` it uses the opposite like US of thousands separator ',' and decimcal Separator '.' so my JDK 11 and 17 thinks `it_IT` is just like the `en_US` for number format while it knows Germany is different.

**Reply by itnoldev** [reply]
In the test you reproduced, the decimal separator doesn't work from the numpad. I run it with JDK 11.
My project is running with the JDK 17

**Reply by melloware** [reply]
It looks like a bug in AutoNumeric: https://github.com/autoNumeric/autoNumeric/issues/602  It says it was fixed but it clearly doesn't work if you use the Numpad.

**Reply by melloware** [reply, marked as answer]
Actually it looks like you need to set http://autonumeric.org/guide#decimalCharacterAlternative which lets the Numeric Keypad work. 

You can test it here: https://codepen.io/AnotherLinuxUser/pen/BaqJOVO?editors=1011

So on your inputNumber set `decimalSeparatorAlternative="."` and it should work.

**Reply by itnoldev** [reply]
Great!
Thank you!
